lib/src/generator/templates.dart

```diff
@@ -32,13 +35,20 @@ class $className {
     final localeName = Intl.canonicalizedLocale(name);${otaEnabled ? '\n${_generateMetadataSetter()}' : ''} 
     return initializeMessages(localeName).then((_) {
       Intl.defaultLocale = localeName;
-      $className.current = $className();
-      
-      return $className.current!;
+      final delegate = $className();
+      $className._current = delegate;
+ 
+      return delegate;
     });
   } 
 
-  static $className? of(BuildContext context) {
+  static $className of(BuildContext context) {
+    final delegate = $className.maybeOf(context);
```

Comment:
Maybe we should rename the `delegate` variable to something else (e.g. `instance`) as it is a bit misleading? 😕